test/unit/org/apache/cassandra/index/sai/SAITester.java

```diff
@@ -239,6 +242,68 @@ public static void setUpClass()
         CQLTester.enableCoordinatorExecution();
     }
 
+    // Pattern that treats apostrophes within words as part of the word
+    protected static final Pattern PATTERN = Pattern.compile("[^\\w']+|'(?=\\s)|(?<=\\s)'");
+    protected static final int DATASET_BODY_COLUMN = 3;
+
+    protected final static Object[][] DATASET =
```

Comment:
These probably are too test-specific constants for the very generic `SAITester`. I'd suggest leaving them in their original `BM25Test` and simply calling them from `FeaturesVersionSupportTest`. 

The constants can be made `public` and imported from `FeaturesVersionSupportTest`. The `insertCollectionData` method can be made static this way:
```
public static void insertCollectionData(SAITester tester)
```
So `FeaturesVersionSupportTest` can call it with:
```
BM25Test.insertCollectionData(this);
```

Reply:
The problem with the call to a non-static method can be avoided with the approach above, adding a `SAITester` argument .